modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/topology/LogicalTopologyImpl.java

```diff
@@ -96,6 +96,10 @@ public void putNode(LogicalNode nodeToPut) {
             // This is an update. First simulate disappearance, then appearance will be fired.
             snapshot = new LogicalTopologySnapshot(snapshot.version() + 1, mapByName.values());
 
+            if (LOG.isInfoEnabled()) {
```

Comment:
I would like to comment the code above.
```
            if (oldNode.id().equals(nodeToPut.id())) {
                // We already have this node, nothing needs to be changed.
                return;
            }
```
How is it possible?

Reply:
It is possible because a node may try to re-join (e.g. after CMG leader change)

Reply:
This is to make this Raft command idempotent